File: ios/Sources/OberonTUI/Palette.swift
// IBM 3270-style 8-colour palette. Index matches TUI.ColorBlack..
// ColorWhite; -1 means "terminal default" and falls back to the
// terminal's foreground/background. Matches web/src/tui-shim.ts so
// programs render identically across hosts.

import SwiftUI

public enum Palette {
    public static let colors: [Color] = [
        Color(red: 0.00, green: 0.00, blue: 0.00),   // 0 Black
        Color(red: 0.00, green: 0.31, blue: 1.00),   // 1 Blue
        Color(red: 1.00, green: 0.19, blue: 0.19),   // 2 Red
        Color(red: 1.00, green: 0.40, blue: 0.80),   // 3 Pink
        Color(red: 0.00, green: 0.80, blue: 0.40),   // 4 Green
        Color(red: 0.00, green: 0.80, blue: 0.80),   // 5 Turquoise
        Color(red: 1.00, green: 0.82, blue: 0.00),   // 6 Yellow
        Color(red: 0.94, green: 0.94, blue: 0.94),   // 7 White
    ]

    /// Map a palette index to a SwiftUI Color, or `nil` for default
    /// (caller substitutes its own foreground/background).
    public static func color(_ i: Int8) -> Color? {
        guard i >= 0, i < 8 else { return nil }
        return colors[Int(i)]
    }
}

File: android/src/main/kotlin/com/oneav/oberon/tui/Palette.kt
// IBM 3270-style 8-colour palette. Index matches TUI.ColorBlack..
// ColorWhite; -1 means "terminal default" — Compose callers
// substitute their own foreground/background. Matches the wasm
// tui-shim's PALETTE and the iOS Palette so programs render
// identically across all three hosts.

package com.oneav.oberon.tui

import androidx.compose.ui.graphics.Color

object Palette {
    val colors: List<Color> = listOf(
        Color(0xFF000000),  // 0 Black
        Color(0xFF0050FF),  // 1 Blue
        Color(0xFFFF3030),  // 2 Red
        Color(0xFFFF66CC),  // 3 Pink
        Color(0xFF00CC66),  // 4 Green
        Color(0xFF00CCCC),  // 5 Turquoise
        Color(0xFFFFD000),  // 6 Yellow
        Color(0xFFF0F0F0),  // 7 White
    )

    fun color(i: Int): Color? = if (i in 0..7) colors[i] else null

    val defaultFg: Color = colors[4]  // ColorGreen — phosphor terminal look
    val defaultBg: Color = colors[0]  // Black
}
